pulpcore/tests/functional/api/test_tasking.py

```diff
@@ -31,8 +36,23 @@ def _dispatch_task(*args, **kwargs):
     return _dispatch_task
 
 
+@pytest.fixture
+def task(dispatch_task, tasks_api_client):
+    """Fixture containing a Task."""
+    task_href = dispatch_task("time.sleep", args=(0,))
+    task = tasks_api_client.read(task_href)
+
+    yield task
+
+    if task.state in TASK_FINAL_STATES:
+        tasks_api_client.delete(task_href)
+    else:
+        tasks_api_client.tasks_cancel(task_href, {"state": "canceled"})
+        tasks_api_client.delete(task_href)
```

Comment:
But also it's fine for tests to leave tasks around...
```suggestion
    if task.state not in TASK_FINAL_STATES:
        tasks_api_client.tasks_cancel(task_href, {"state": "canceled"})
        # here may be another monitor_task needed, as cancelling is only triggered by that call.
    tasks_api_client.delete(task_href)
```

Reply:
Also yet another thought: we could put that cleanup at the end of the dispatch_task fixture.